rdt-open-reader/src/main/java/io/ona/rdt/util/Utils.java

```diff
@@ -263,7 +265,11 @@ private UserAuthorizationVerificationTask(Context context) {
         @Override
         protected Void doInBackground(Void... voids) {
             boolean isUserAuthorized = syncUtils.verifyAuthorization();
-            if (!isUserAuthorized) {
+
+            Activity covidJsonFormActivity = RDTApplication.getInstance().getCurrentActivity();
+            boolean isFormActivity = covidJsonFormActivity != null && CovidJsonFormActivity.class.getSimpleName().equals(covidJsonFormActivity.getClass().getSimpleName());
```

Comment:
I think we can check for `currentActivity != null &&  currentActivity instanceof CovidJsonFormActivity` instead.

Reply:
fixed.